### ISSUE TITLE
Error while running example.js  

### ISSUE THREAD

**Issue by ff6347** [automat/sketch-context2d#1]
Hi @automat,

when I found your module I got pretty excited. So I wanted to try it out.But I get an error when running it.

I installed your module with npm by running:

```
npm install git+git@github.com:automat/sketch-context2d.git -S
```

copied the example sketch into my index.js file and ran the script:  

```
./node_modules/.bin/sketch-context2d index.js
```

Unfortunately I get the following error:  

```
/path/to/folder/node_modules/sketch-context2d/index.js:118
            throw new Error(err);
            ^

Error: SyntaxError: Unexpected token (94:82) while parsing file: /path/to/folder/node_modules/sketch-context2d/index.js
at Readable.<anonymous> (/path/to/folder/node_modules/sketch-context2d/index.js:118:23)
at emitOne (events.js:96:13)
at Readable.emit (events.js:188:7)
at Labeled.<anonymous> (/path/to/folder/node_modules/read-only-stream/index.js:28:44)
at emitOne (events.js:96:13)
at Labeled.emit (events.js:188:7)
at Labeled.<anonymous> (/path/to/folder/node_modules/stream-splicer/index.js:130:18)
at emitOne (events.js:101:20)
at Labeled.emit (events.js:188:7)
at Deps.<anonymous> (/path/to/folder/node_modules/stream-splicer/index.js:130:18)
```

**Comment by automat** [comment]
Hi,

here are some hints which might solve your problem:

`sketch-context-2d` is currently _not compatible_ with the latest sketch version, its build against `version 3.5.2`, you can grab it here www.sketchapp.com/download/sketch-3.5.2.zip

You will need to run the cmd manually using `./node-modules/sketch-context-2d/bin/cmd.js absolute/path/to/your/file.js --verbose`, as there are some permission problems running the `npm bin cmd`.

Try running this as a minimal test:

```
function main(canvas){
    const ctx = canvas.getContext('2d');
    console.log('artboard size', canvas.width, canvas.height);
    ctx.strokeStyle = '#000';
    ctx.moveTo(0,0);
    ctx.lineTo(canvas.width, canvas.height);
    ctx.stroke();
} 
module.exports = main;
```

Hope ill find some time in the next weeks to update to `Sketch 3.8.*` + hope this helps.

**Comment by ff6347** [comment]
Hi. I used the latest sketch version and used the file in .bin. This might be the problem. The example.js was the minimal example you provided in the README.md on GitHub. I'll give it a try later on. Thanks

**Comment by ff6347** [comment]
- I downloaded sketch 3.5.2
- I renamed it to Sketch.3.5.2.app to have both versions)  
- copy pasted your example code from the issue
- ran it

command:  

```
$ ./node_modules/sketch-context2d/bin/cmd.js /Users/me/Desktop/sketch/index.js --verbose
```

The result is:  

```
/Users/me/Desktop/sketch/node_modules/sketch-context2d/index.js:51
        throw new Error(err || stderr);
        ^

Error: 2016-07-14 21:02:24.613 coscript[54580:1397434] Could not find a JSTalk connection to com.bohemiancoding.sketch3.JSTalk
2016-07-14 21:02:24.625 coscript[54580:1397434] Exception: {
    column = 25;
    line = 1;
    sourceURL = "/Users/me/Desktop/sketch/-e";
}

at /Users/me/Desktop/sketch/node_modules/sketch-context2d/index.js:51:19
at ChildProcess.exithandler (child_process.js:193:7)
at emitTwo (events.js:106:13)
at ChildProcess.emit (events.js:191:7)
at maybeClose (internal/child_process.js:852:16)
at Socket.<anonymous> (internal/child_process.js:323:11)
at emitOne (events.js:96:13)
at Socket.emit (events.js:188:7)
at Pipe._handle.close [as _onclose] (net.js:492:12)
```
- renamed current sketch version
- named sketch 3.5.2 to Sketch.app

Still no luck with running it. :-(

My system is:  
MacBook Air 13" Mid 2013  
Distro: OS X 10.11.5
Node: v6.3.0

**Comment by automat** [comment]
Hm, 
ok i see.

You installed Sketch `3.8` first, right? Both `3.8` and `3.5.2` share the same bundle identifier `com.bohemiancoding.sketch3`.`sketch-context2d` is using `COScript` to start Sketch.

https://github.com/ccgus/CocoaScript/blob/master/src/framework/COScript.m#L561 
Gets bundle identifier from app path which defaults to just `Sketch`
https://github.com/automat/sketch-context2d/blob/master/index.js#L48
`COScript` connects to `Sketch` jstalk running process
https://github.com/ccgus/CocoaScript/blob/master/src/framework/COScript.m#L589
So basically it starts the latest `Sketch` installed and looks for the process with 
the identifier, if its not already running.

I removed that (see d09568e). Its now just looking at running processes. This might not solve you problem directly, but will prevent auto-starting the wrong version.

Reinstall `sketch-context2d` in your test-project and give it a try with just having version `3.5.2` open. Paste your logs here again.

**Comment by ff6347** [comment]
Before restarting the computer I got the same error as before. After a restart I get this:  

```
Error: Unable to load Context2d.framework
line: 9
sourceURL: /Users/icke/Desktop/p5-sketch/node_modules/sketch-context2d/plugin/plugin.cocoascript
column: 24
```

**Comment by automat** [comment]
Hi,

i finished the initial update to `Sketch 3.8`. Everything should work with the new version now, except some minor deprecated api warnings, which i will fix in the upcoming weeks.

If you want to give it a try, clone the `Sketch 3.8` branch https://github.com/automat/sketch-context2d/tree/sketch.3.8, link the module temporarily using `npm link` and try running the minimal test described above or one of the tests in `/test/CanvasRenderingContext2d-API` using `sketch-context2d path/to/file.js --verbose`.

**Comment by ff6347** [comment]
Awesome.

 Unfortunately it crashes Sketch on my machine. 
 When there is no artboard It tells me that there is none. If there is I get this error:  

```
/Users/icke/Desktop/sketch-context2d/index.js:61
            throw new Error(err || stderr);
            ^

Error: 2016-07-20 08:36:26.930 coscript[5621:64591] Exception: (null)

    at /Users/icke/Desktop/sketch-context2d/index.js:61:19
    at ChildProcess.exithandler (child_process.js:193:7)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at maybeClose (internal/child_process.js:852:16)
    at Socket.<anonymous> (internal/child_process.js:323:11)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7)
    at Pipe._handle.close [as _onclose] (net.js:492:12)
```

I cloned the 3.8 branch:  

```
git clone -b sketch.3.8 git@github.com:automat/sketch-context2d.git
```

Linked it:  

```
cd sketch-context2d/
npm link
```

Ran the minimal example (index.js):  

```
function main(canvas){
    const ctx = canvas.getContext('2d');
    console.log('artboard size', canvas.width, canvas.height);
    ctx.strokeStyle = '#000';
    ctx.moveTo(0,0);
    ctx.lineTo(canvas.width, canvas.height);
    ctx.stroke();
}
module.exports = main;
```

like this:  

```
sketch-context2d index.js --verbose
```

And also with an absolut path

```
sketch-context2d /Users/me/folder/index.js --verbose
```

No luck. (Using sketch 3.8.3)  

I attached the Sketch crash report. Maybe that helps you. Another thing I really would love to see is using your module as a local dependency. If you want me to run another test let me know. Still looking forward to use it. :-)

[sketch-context2d-crash-report.txt](https://github.com/automat/sketch-context2d/files/372990/sketch-context2d-crash-report.txt)

**Comment by automat** [comment]
Looks like it is still calling a deprecated method, `addNewStylePart`. I did not push the latest library build. Pull again and give it another try. Commit 9471cc8

**Comment by ff6347** [comment]
YEY! ✨💫 🎁  🎉 🎊 💃 🎋

```
Wed Jul 20 2016 09:56:47 GMT+0200 (CEST)
Script start
artboard size,1000,1000
Script end: 00:00:00
```

awesome!

**Comment by ff6347** [comment]
BTW if you want me to run some tests for you let me know.

**Comment by automat** [comment]
You can find a complete list of single tests for the CanvasRenderingContext2d-API in https://github.com/automat/sketch-context2d/tree/master/test/CanvasRenderingContext2d-API. Theres also a support table showing the current implementation status: https://github.com/automat/sketch-context2d/blob/master/test/CanvasRenderingContext2d-API/SUMMARY.md. It would be great if you can test the ones marked as _implemented_ and if you find something open an issue.

That would be really helpful, thanks!

**Comment by ff6347** [comment]
I will ASAP

**Comment by ff6347** [comment]
i tested all the scripts marked as implemented. They ran without errors except for the two I created issues for. What I did not check is the content of the scripts. So I cant tell if they did what they where supposed to do. 
For this I need some more time that I don't have

**Comment by automat** [comment]
No problem. Thanks for your help so far. Both issues reported are caused by api changes in `Sketch 3.9`. I will update to the new version within the next few days and implement the canvas layer composition functionality. Will release a first version after that. Thanks again

**Comment by ff6347** [comment]
Your very welcome. Looking forward to the first release
